std/stdio.d

```diff
@@ -725,7 +725,8 @@ $(D rawRead) always reads in binary mode on Windows.
     {
         import std.exception : enforce, errnoEnforce;
 
-        enforce(buffer.length, "rawRead must take a non-empty buffer");
+        if (!buffer.length)
+            enforce(false, "rawRead must take a non-empty buffer");
```

Comment:
dmd does not inline functions with lazy parameters. It's a big problem for enforce, been around forever.